actionview/lib/action_view/helpers/date_helper.rb

```diff
@@ -97,55 +97,37 @@ def distance_of_time_in_words(from_time, to_time = 0, options = {})
           scope: :'datetime.distance_in_words'
         }.merge!(options)
 
-        from_time = normalize_distance_of_time_argument_to_time(from_time)
-        to_time = normalize_distance_of_time_argument_to_time(to_time)
-        from_time, to_time = to_time, from_time if from_time > to_time
-        distance_in_minutes = ((to_time - from_time) / 60.0).round
-        distance_in_seconds = (to_time - from_time).round
+        distance = DateDistance.new(from_time, to_time)
 
         I18n.with_options locale: options[:locale], scope: options[:scope] do |locale|
-          case distance_in_minutes
-          when 0..1
-            return distance_in_minutes == 0 ?
+          if distance.in_minutes <= 1
+            return distance.in_minutes == 0 ?
                    locale.t(:less_than_x_minutes, count: 1) :
-                   locale.t(:x_minutes, count: distance_in_minutes) unless options[:include_seconds]
+                   locale.t(:x_minutes, count: distance.in_minutes) unless options[:include_seconds]
 
-            case distance_in_seconds
+            case distance.in_seconds
             when 0..4   then locale.t :less_than_x_seconds, count: 5
             when 5..9   then locale.t :less_than_x_seconds, count: 10
             when 10..19 then locale.t :less_than_x_seconds, count: 20
             when 20..39 then locale.t :half_a_minute
             when 40..59 then locale.t :less_than_x_minutes, count: 1
             else             locale.t :x_minutes,           count: 1
             end
-
-          when 2...45           then locale.t :x_minutes,      count: distance_in_minutes
-          when 45...90          then locale.t :about_x_hours,  count: 1
-            # 90 mins up to 24 hours
-          when 90...1440        then locale.t :about_x_hours,  count: (distance_in_minutes.to_f / 60.0).round
-            # 24 hours up to 42 hours
```

Comment:
The `when 45...90` case can be removed because `(45...90).map{|i| (i.to_f / 60.0).round }.uniq => [1]`